From 3f885794e11b02c09c1d3d2768c696255ca7169a Mon Sep 17 00:00:00 2001
Message-Id: <3f885794e11b02c09c1d3d2768c696255ca7169a.1677692382.git.stefan@agner.ch>
In-Reply-To: <35714823562dc6d7df8b8e81747aa19e27b9d5f7.1677692382.git.stefan@agner.ch>
References: <35714823562dc6d7df8b8e81747aa19e27b9d5f7.1677692382.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Thu, 12 Jan 2023 14:35:04 +0100
Subject: [PATCH] [rest] explicitly set Connection header to close

By default, HTTP 1.1 connections should stay open after a transaction.
However, that is not how the current REST server implementation behaves:
After each transaction the HTTP connection is being closed by the
server.

Set the Connection header to "close" to tell the client about this
behavior.
---
 src/rest/response.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/rest/response.cpp b/src/rest/response.cpp
index 540a0840e0..5c08fd77b4 100644
--- a/src/rest/response.cpp
+++ b/src/rest/response.cpp
@@ -35,6 +35,7 @@
     "Access-Control-Allow-Headers, Origin,Accept, X-Requested-With, Content-Type, Access-Control-Request-Method, " \
     "Access-Control-Request-Headers"
 #define OT_REST_RESPONSE_ACCESS_CONTROL_ALLOW_METHOD "GET, OPTIONS, POST, PUT"
+#define OT_REST_RESPONSE_CONNECTION "close"
 
 namespace otbr {
 namespace rest {
@@ -51,6 +52,7 @@ Response::Response(void)
     mHeaders["Access-Control-Allow-Origin"]  = OT_REST_RESPONSE_ACCESS_CONTROL_ALLOW_ORIGIN;
     mHeaders["Access-Control-Allow-Methods"] = OT_REST_RESPONSE_ACCESS_CONTROL_ALLOW_METHOD;
     mHeaders["Access-Control-Allow-Headers"] = OT_REST_RESPONSE_ACCESS_CONTROL_ALLOW_HEADERS;
+    mHeaders["Connection"]                   = OT_REST_RESPONSE_CONNECTION;
 }
 
 void Response::SetComplete()
-- 
2.39.1

